### PR TITLE
fix: mitigate integer overflow issues in x/liquidity

### PR DESCRIPTION
**Pull request by hallazzang**
## Description

This PR adds restrictions around coin amount to mitigate integer overflow issues that could cause panic on chain.

## Tasks

- [x] Define `MinCoinAmount`, `MaxCoinAmount`
- [x] Add `SafeMath` panic guard

### CODE REVIEW

**Review by kogisin — APPROVED** [review]
Great work! LGTM. `SafeMath` and `IsOverflow` functions are going to be very useful.

File: x/liquidity/amm/pool.go [hunk 1]
```diff
@@ -124,11 +136,19 @@ func (pool *BasicPool) LowestSellPrice() (price sdk.Dec, found bool) {
...
 		return sdk.ZeroInt()
 	}
-	return pool.rx.ToDec().QuoTruncate(price).Sub(pool.ry.ToDec()).TruncateInt()
+	utils.SafeMath(func() {
+		amt = pool.rx.ToDec().QuoTruncate(price).Sub(pool.ry.ToDec()).TruncateInt()
+		if amt.GT(MaxCoinAmount) {
+			amt = MaxCoinAmount
+		}
+	}, func() {
+		amt = MaxCoinAmount
```
Comment (dongsam) [inline x/liquidity/amm/pool.go:149]: Don't we need to add this logic on spec? If someone reach this logic without knowing this fact, someone will likely suffer losses

Comment (hallazzang) [reply x/liquidity/amm/pool.go:149]: Can you explain more about what *losses* means here?

File: x/liquidity/keeper/pool_test.go [hunk 2]
```diff
@@ -250,6 +250,14 @@ func (s *KeeperTestSuite) TestDepositRefundTooSmallMintedPoolCoin() {
...
 }
 
+func (s *KeeperTestSuite) TestTooLargePool() {
+	pair := s.createPair(s.addr(0), "denom1", "denom2", true)
+	pool := s.createPool(s.addr(0), pair.Id, utils.ParseCoins("1000000denom1,1000000denom2"), true)
+
+	_, err := s.keeper.Deposit(s.ctx, types.NewMsgDeposit(s.addr(1), pool.Id, utils.ParseCoins("10000000000000000000000000000000000000000denom1,10000000000000000000000000000000000000000denom2")))
+	s.Require().ErrorIs(err, types.ErrTooLargePool)
+}
+
```
Comment (hallazzang) [inline x/liquidity/keeper/pool_test.go:260]: Add `SendCoins` case

**Review by dongsam — APPROVED** [review]

**Pull request merged**